gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -203,8 +203,8 @@ protected void submitFlowToDagManagerHelper(String flowGroup, String flowName, S
       URI flowUri = FlowSpec.Utils.createFlowSpecUri(flowId);
       spec = (FlowSpec) flowCatalog.getSpecs(flowUri);
       // Adds flowExecutionId to config to ensure they are consistent across hosts
-      FlowSpec updatedSpec = FlowSpec.Utils.createFlowSpecWithProperty(spec, ConfigurationKeys.FLOW_EXECUTION_ID_KEY, flowExecutionId);
-      this.orchestrator.submitFlowToDagManager(updatedSpec);
+      spec.addPropertyToConfigAsProperties(ConfigurationKeys.FLOW_EXECUTION_ID_KEY, flowExecutionId);
```

Comment:
can you point me to the code where we query this flow execution id?

Reply:
Yes we query here from `ConfigAsProperties` https://github.com/apache/gobblin/blob/9e30c6c3bca857942fba5d10345b1be14adc1942/gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java#L186

Reply:
this is where we add flow.execution.id
in li-gobblin AzkabanHeadlessProducer we get the configs 
`Map<String, String> flowParameters = new HashMap<>(Maps.fromProperties(ConfigUtils.configToProperties(jobConfig)));`
and there we use configs from `config` object. so you have to make change there also, if you want this approach to work. lets think more thoroughly which approach is better and more intuitive.